pkg/services/publicdashboards/service/service.go

```diff
@@ -206,7 +206,17 @@ func (pd *PublicDashboardServiceImpl) GetQueryDataResponse(ctx context.Context,
 		return nil, err
 	}
 
-	return pd.QueryDataService.QueryDataMultipleSources(ctx, anonymousUser, skipCache, metricReq, true)
+	res, err := pd.QueryDataService.QueryDataMultipleSources(ctx, anonymousUser, skipCache, metricReq, true)
+
+	// We want to track which datasources were successful and which were not
+	reqDatasources := metricReq.GetUniqueDatasourceTypes()
+	if err != nil {
+		pd.log.Error("Error querying datasources for public dashboard", "error", err.Error(), "datasources", reqDatasources)
```

Comment:
Is there a preferred format for logging an array?

Reply:
This seems ok for the log but how is it shown in the error message?  Are the values separated by spaces?

Reply:
Looks like this:

`INFO [09-13|16:34:59] Successfully queried datasources for public dashboard logger=publicdashboards.service datasources="[prometheus postgres]"`